modules/consul/consul.go

```diff
@@ -0,0 +1,88 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultHttpApiPort = "8500"
+	defaultBrokerPort  = "8600"
+)
+
+const (
+	DefaultBaseImage = "docker.io/hashicorp/consul:1.15"
+)
+
+// ConsulContainer represents the Consul container type used in the module.
+type ConsulContainer struct {
+	testcontainers.Container
+}
+
+// ApiEndpoint returns host:port for the HTTP API endpoint.
+func (c *ConsulContainer) ApiEndpoint(ctx context.Context) (string, error) {
+	mappedPort, err := c.MappedPort(ctx, defaultHttpApiPort)
+	if err != nil {
+		return "", err
+	}
+
+	hostIP, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	uri := fmt.Sprintf("%s:%s", hostIP, mappedPort.Port())
+	return uri, nil
+}
+
+// WithConfigString takes in a JSON string of keys and values to define a configuration to be used by the instance.
+func WithConfigString(config string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["CONSUL_LOCAL_CONFIG"] = config
+	}
+}
+
+// WithConfigFile takes in a path to a JSON file to define a configuration to be used by the instance.
+func WithConfigFile(configPath string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configPath,
+			ContainerFilePath: "/consul/config/node.json",
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+	}
+}
+
+// RunContainer creates an instance of the Consul container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*ConsulContainer, error) {
+	containerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image: DefaultBaseImage,
+			ExposedPorts: []string{
+				defaultHttpApiPort + "/tcp",
+				defaultBrokerPort + "/tcp",
+				defaultBrokerPort + "/udp",
+			},
+			Env: map[string]string{},
+			WaitingFor: wait.ForAll(
+				wait.ForLog("Consul agent running!"),
```

Comment:
note this introduced a flake. The former example was waiting for the listen port (as are most ad-hoc usages on github). This passes before the api is ready failing tests. I think it should be reverted to what the example used:


```
		testcontainers.WithWaitStrategy(wait.ForListeningPort("8500/tcp")),
```

Reply:
@codefromthecrypt I went ahead and created an issue for it. Would you like to work on the fix?

Reply:
actually yes thanks!

Reply:
Completely an oversight on my part, thank you for noticing!

Reply:
No worries @Chris-AS1 shit happens 😄 Thankfully this community has eyes on every place 💪